### PR TITLE
Update HMR Client Runtime

### PR DESCRIPTION
**Pull request by Timer**
This pull request:

- Pulls in the latest changes from Create React App's HMR client
- Updates the `typeChecked` event handling for improved reliability (should make the TypeScript HMR tests stop failing)

Let's see if all of our HMR tests pass 🤞!

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| buildDuration | 10.8s | 10.5s | -226ms |
| nodeModulesSize | 55.5 MB | 55.5 MB | ⚠️ +351 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 1.851 | 1.798 | -0.05 |
| / avg req/sec | 1350.94 | 1390.47 | ⚠️ +39.53 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 0.943 | 0.946 | 0 |
| /error-in-render avg req/sec | 2650 | 2643.2 | -6.8 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.25 kB | 6.25 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..5e5e.js gzip | 10.3 kB | 10.3 kB | ✓ |
| framework.a1..NSE.txt gzip | 220 B | 220 B | ✓ |
| framework.a1..NSE.txt gzip | 220 B | 220 B | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| Overall change | 56.7 kB | 56.7 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.79 kB | 4.79 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.85 kB | 6.85 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| Overall change | 51.4 kB | 51.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.2 kB | 26.2 kB | ✓ |
| Overall change | 26.2 kB | 26.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| _app.js gzip | 1.24 kB | 1.24 kB | ✓ |
| _error.js gzip | 3.12 kB | 3.12 kB | ✓ |
| hooks.js gzip | 664 B | 664 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.06 kB | 2.06 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 7.86 kB | 7.86 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| _app.module.js gzip | 602 B | 602 B | ✓ |
| _error.module.js gzip | 2.09 kB | 2.09 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.38 kB | 5.38 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| index.html gzip | 929 B | 929 B | ✓ |
| link.html gzip | 938 B | 938 B | ✓ |
| withRouter.html gzip | 927 B | 927 B | ✓ |
| Overall change | 2.79 kB | 2.79 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| buildDuration | 12s | 12s | -19ms |
| nodeModulesSize | 55.5 MB | 55.5 MB | ⚠️ +351 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.25 kB | 6.25 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..5e5e.js gzip | 10.3 kB | 10.3 kB | ✓ |
| framework.a1..NSE.txt gzip | 220 B | 220 B | ✓ |
| framework.a1..NSE.txt gzip | 220 B | 220 B | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| Overall change | 56.7 kB | 56.7 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.79 kB | 4.79 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.85 kB | 6.85 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| Overall change | 51.4 kB | 51.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.2 kB | 26.2 kB | ✓ |
| Overall change | 26.2 kB | 26.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| _app.js gzip | 1.24 kB | 1.24 kB | ✓ |
| _error.js gzip | 3.12 kB | 3.12 kB | ✓ |
| hooks.js gzip | 664 B | 664 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.06 kB | 2.06 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 7.86 kB | 7.86 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| _app.module.js gzip | 602 B | 602 B | ✓ |
| _error.module.js gzip | 2.09 kB | 2.09 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.38 kB | 5.38 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| _error.js | 558 kB | 558 kB | ✓ |
| 404.html | 4.18 kB | 4.18 kB | ✓ |
| hooks.html | 3.81 kB | 3.81 kB | ✓ |
| index.js | 558 kB | 558 kB | ✓ |
| link.js | 594 kB | 594 kB | ✓ |
| routerDirect.js | 587 kB | 587 kB | ✓ |
| withRouter.js | 587 kB | 587 kB | ✓ |
| Overall change | 2.89 MB | 2.89 MB | ✓ |

</details>

</details>

**Comment by ijjk** [comment]
## Failing test suites 

**test/integration/typescript-hmr/test/index.test.js** 
- TypeScript HMR > should recover from a type error

<details>
<summary>Expand output</summary>

  ● TypeScript HMR › should recover from a type error

    TIMED OUT: /success/

    fail

      336 |     }
      337 |     console.error('TIMED OUT CHECK: ', { regex, content })
    > 338 |     throw new Error('TIMED OUT: ' + regex + '\n\n' + content)
          |           ^
      339 |   }, 1000 * 30)
      340 |   while (!found) {
      341 |     try {

      at Timeout._onTimeout (lib/next-test-utils.js:338:11)

</details>

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| buildDuration | 10.2s | 10.4s | ⚠️ +239ms |
| nodeModulesSize | 55.5 MB | 55.5 MB | ⚠️ +626 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 1.7 | 1.755 | ⚠️ +0.05 |
| / avg req/sec | 1470.51 | 1424.45 | -46.06 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 0.893 | 0.91 | ⚠️ +0.02 |
| /error-in-render avg req/sec | 2799.39 | 2745.77 | -53.62 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.25 kB | 6.25 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..5e5e.js gzip | 10.3 kB | 10.3 kB | ✓ |
| framework.a1..NSE.txt gzip | 220 B | 220 B | ✓ |
| framework.a1..NSE.txt gzip | 220 B | 220 B | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| Overall change | 56.7 kB | 56.7 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.79 kB | 4.79 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.85 kB | 6.85 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| Overall change | 51.4 kB | 51.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.2 kB | 26.2 kB | ✓ |
| Overall change | 26.2 kB | 26.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| _app.js gzip | 1.24 kB | 1.24 kB | ✓ |
| _error.js gzip | 3.12 kB | 3.12 kB | ✓ |
| hooks.js gzip | 664 B | 664 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.06 kB | 2.06 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 7.86 kB | 7.86 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| _app.module.js gzip | 602 B | 602 B | ✓ |
| _error.module.js gzip | 2.09 kB | 2.09 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.38 kB | 5.38 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| index.html gzip | 929 B | 929 B | ✓ |
| link.html gzip | 938 B | 938 B | ✓ |
| withRouter.html gzip | 927 B | 927 B | ✓ |
| Overall change | 2.79 kB | 2.79 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| buildDuration | 11s | 11.4s | ⚠️ +401ms |
| nodeModulesSize | 55.5 MB | 55.5 MB | ⚠️ +626 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.25 kB | 6.25 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..5e5e.js gzip | 10.3 kB | 10.3 kB | ✓ |
| framework.a1..NSE.txt gzip | 220 B | 220 B | ✓ |
| framework.a1..NSE.txt gzip | 220 B | 220 B | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| Overall change | 56.7 kB | 56.7 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.79 kB | 4.79 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.85 kB | 6.85 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| Overall change | 51.4 kB | 51.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.2 kB | 26.2 kB | ✓ |
| Overall change | 26.2 kB | 26.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| _app.js gzip | 1.24 kB | 1.24 kB | ✓ |
| _error.js gzip | 3.12 kB | 3.12 kB | ✓ |
| hooks.js gzip | 664 B | 664 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.06 kB | 2.06 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 7.86 kB | 7.86 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| _app.module.js gzip | 602 B | 602 B | ✓ |
| _error.module.js gzip | 2.09 kB | 2.09 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.38 kB | 5.38 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| _error.js | 558 kB | 558 kB | ✓ |
| 404.html | 4.18 kB | 4.18 kB | ✓ |
| hooks.html | 3.81 kB | 3.81 kB | ✓ |
| index.js | 558 kB | 558 kB | ✓ |
| link.js | 594 kB | 594 kB | ✓ |
| routerDirect.js | 587 kB | 587 kB | ✓ |
| withRouter.js | 587 kB | 587 kB | ✓ |
| Overall change | 2.89 MB | 2.89 MB | ✓ |

</details>

</details>

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| buildDuration | 12.7s | 13.1s | ⚠️ +405ms |
| nodeModulesSize | 55.5 MB | 55.5 MB | ⚠️ +626 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.209 | 2.229 | ⚠️ +0.02 |
| / avg req/sec | 1131.56 | 1121.79 | -9.77 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.292 | 1.27 | -0.02 |
| /error-in-render avg req/sec | 1934.74 | 1968.65 | ⚠️ +33.91 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.25 kB | 6.25 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..5e5e.js gzip | 10.3 kB | 10.3 kB | ✓ |
| framework.a1..NSE.txt gzip | 220 B | 220 B | ✓ |
| framework.a1..NSE.txt gzip | 220 B | 220 B | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| Overall change | 56.7 kB | 56.7 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.79 kB | 4.79 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.85 kB | 6.85 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| Overall change | 51.4 kB | 51.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.2 kB | 26.2 kB | ✓ |
| Overall change | 26.2 kB | 26.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| _app.js gzip | 1.24 kB | 1.24 kB | ✓ |
| _error.js gzip | 3.12 kB | 3.12 kB | ✓ |
| hooks.js gzip | 664 B | 664 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.06 kB | 2.06 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 7.86 kB | 7.86 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| _app.module.js gzip | 602 B | 602 B | ✓ |
| _error.module.js gzip | 2.09 kB | 2.09 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.38 kB | 5.38 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| index.html gzip | 929 B | 929 B | ✓ |
| link.html gzip | 938 B | 938 B | ✓ |
| withRouter.html gzip | 927 B | 927 B | ✓ |
| Overall change | 2.79 kB | 2.79 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| buildDuration | 13.8s | 13.7s | -31ms |
| nodeModulesSize | 55.5 MB | 55.5 MB | ⚠️ +626 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.25 kB | 6.25 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..5e5e.js gzip | 10.3 kB | 10.3 kB | ✓ |
| framework.a1..NSE.txt gzip | 220 B | 220 B | ✓ |
| framework.a1..NSE.txt gzip | 220 B | 220 B | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| Overall change | 56.7 kB | 56.7 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.79 kB | 4.79 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.85 kB | 6.85 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| Overall change | 51.4 kB | 51.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.2 kB | 26.2 kB | ✓ |
| Overall change | 26.2 kB | 26.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| _app.js gzip | 1.24 kB | 1.24 kB | ✓ |
| _error.js gzip | 3.12 kB | 3.12 kB | ✓ |
| hooks.js gzip | 664 B | 664 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.06 kB | 2.06 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 7.86 kB | 7.86 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| _app.module.js gzip | 602 B | 602 B | ✓ |
| _error.module.js gzip | 2.09 kB | 2.09 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.38 kB | 5.38 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | zeit/next.js canary  | Timer/next.js update/hot-dev-client | Change |
| - | - | - | - |
| _error.js | 558 kB | 558 kB | ✓ |
| 404.html | 4.18 kB | 4.18 kB | ✓ |
| hooks.html | 3.81 kB | 3.81 kB | ✓ |
| index.js | 558 kB | 558 kB | ✓ |
| link.js | 594 kB | 594 kB | ✓ |
| routerDirect.js | 587 kB | 587 kB | ✓ |
| withRouter.js | 587 kB | 587 kB | ✓ |
| Overall change | 2.89 MB | 2.89 MB | ✓ |

</details>

</details>

File: packages/next/client/dev/error-overlay/hot-dev-client.js [hunk 1]
```diff
@@ -325,16 +369,12 @@ async function tryApplyUpdates(onHotUpdateSuccess) {
...
 
   // https://webpack.github.io/docs/hot-module-replacement.html#check
-  try {
-    const updatedModules = await module.hot.check(
-      /* autoApply */ {
-        ignoreUnaccepted: true,
-      }
-    )
-    if (updatedModules) {
+  module.hot.check(/* autoApply */ true).then(
```
Comment (ijjk) [inline packages/next/client/dev/error-overlay/hot-dev-client.js:372]: Should we update the above link with latest `https://webpack.js.org/api/hot-module-replacement/#check` current one redirects to home page of `webpack.js.org` after 3 seconds

File: packages/next/client/dev/error-overlay/hot-dev-client.js [hunk 2]
```diff
@@ -325,16 +369,12 @@ async function tryApplyUpdates(onHotUpdateSuccess) {
   }
 
   // https://webpack.github.io/docs/hot-module-replacement.html#check
```
Comment (Timer) [inline packages/next/client/dev/error-overlay/hot-dev-client.js:371]:
```suggestion
  // https://webpack.js.org/api/hot-module-replacement/#check
```

**Review by ijjk — APPROVED** [review]

**Pull request merged**